pyop2/types/mat.py

```diff
@@ -383,10 +381,10 @@ def __init__(self, parent, i, j):
         self._dims = tuple([tuple([parent.dims[i][j]])])
         self._blocks = [[self]]
         self.iteration_regions = parent.iteration_regions
-        self.lcomm = mpi.internal_comm(self.dsets[0].comm)
-        self.rcomm = mpi.internal_comm(self.dsets[1].comm)
+        self.lcomm = mpi.internal_comm(self.dsets[0].comm, self)
```

Comment:
Is it necessary to increment the reference count for the internal comm here? Could we not have something like:
```py
@property
def lcomm(self):
    return self.dsets[0]._comm
```

Reply:
No, but it's easier to use the pattern than to break it in this one instance. The whole `lcomm` and `rcomm` thing seems redundant to me, it isn't actually being used. I think the original code was aspirational and the functionality that would utilise it never implemented in PyOP2. There's a comment to that effect somewhere in the code.

Reply:
Yeah, petsc operators don't have different left and right communicators so this is over-eager and could be culled and replaced with a check in the constructor that the communicators of the row and column sets are the same.